subsystem/cluster_test.go

```diff
@@ -2954,6 +2997,44 @@ var _ = Describe("cluster install, with default network params", func() {
 		Expect(c.InstallCompletedAt).ShouldNot(Equal(startTimeInstalled))
 		Expect(c.InstallCompletedAt).Should(Equal(c.StatusUpdatedAt))
 	})
+	Context("fail disk speed", func() {
+		It("first host", func() {
+			clusterID := *cluster.ID
+			registerHostsAndSetRoles(clusterID, 5)
+			rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
+			Expect(err).NotTo(HaveOccurred())
+			c := rep.GetPayload()
+			startTimeInstalling := c.InstallStartedAt
+
+			c = tryInstallClusterWithDiskResponses(clusterID, c.Hosts[1:], c.Hosts[:1])
+			Expect(len(c.Hosts)).Should(Equal(5))
+			Expect(c.InstallStartedAt).ShouldNot(Equal(startTimeInstalling))
+		})
+		It("all hosts", func() {
+			clusterID := *cluster.ID
+			registerHostsAndSetRoles(clusterID, 5)
+			rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
+			Expect(err).NotTo(HaveOccurred())
+			c := rep.GetPayload()
+			startTimeInstalling := c.InstallStartedAt
+
+			c = tryInstallClusterWithDiskResponses(clusterID, nil, c.Hosts)
+			Expect(len(c.Hosts)).Should(Equal(5))
+			Expect(c.InstallStartedAt).ShouldNot(Equal(startTimeInstalling))
+		})
+		It("last host", func() {
+			clusterID := *cluster.ID
+			registerHostsAndSetRoles(clusterID, 5)
+			rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
+			Expect(err).NotTo(HaveOccurred())
+			c := rep.GetPayload()
+			startTimeInstalling := c.InstallStartedAt
+
+			c = tryInstallClusterWithDiskResponses(clusterID, nil, c.Hosts[len(c.Hosts)-1:])
```

Comment:
Not sure if this is a typo or intended but why not ?
```
tryInstallClusterWithDiskResponses(clusterID, c.Hosts[:len(c.Hosts)-1], c.Hosts[len(c.Hosts)-1:])
```

Reply:
Intended

Reply:
What is wrong with it?

Reply:
Based on the function name I assumed that you want only the last host to fail, so I expected failingHosts=onlyTheLastHost and successHosts=allTheRest.

Isn't it the case?

Reply:
Not exactly.  The last one fails, and others are unknown.

Reply:
OK then, lgtm.
You already have the label